packages/cli/src/gemini.tsx

```diff
@@ -350,8 +364,8 @@ export async function main() {
       // Not in a sandbox and not entering one, so relaunch with additional
       // arguments to control memory usage if needed.
       if (memoryArgs.length > 0) {
-        await relaunchWithAdditionalArgs(memoryArgs);
-        process.exit(0);
+        await relaunchAppInChildProcess(memoryArgs);
```

Comment:
see the start_sandbox call a few lines up above. that also needs the same logic to relaunch on restart. likely you can be clever and avoid duplicating the logic to check the return code but potentially it is easier to special case and add that code to start_sandbox. please test with the sandbox to ensure restarting works there as well.